### ISSUE TITLE
API exits for 4 atoms

### ISSUE THREAD

**Issue by Acetylsalicylsaeure** [openmopac/mopac#230]
https://github.com/openmopac/mopac/blob/48500c009cec6c4167882fa94b111d5c6db58ded/src/geometry/gmetry.F90#L112
this line seems to be causing a forced exit when trying to optimize a geometry with 4 atoms.

```gdb
112	        if(size(na) == 4) call exit(1)
(gdb)
_gfortran_exit_i4 (status=status@entry=0x1555511196a0)
    at ../../../libgfortran/intrinsics/exit.c:38
warning: 38	../../../libgfortran/intrinsics/exit.c: No such file or directory
(gdb) bt
#0  _gfortran_exit_i4 (status=status@entry=0x1555511196a0)
    at ../../../libgfortran/intrinsics/exit.c:38
#1  0x0000155550d805a1 in gmetry (geo=..., coord=...)
    at /home/paul/programs/mopac/src/geometry/gmetry.F90:112
#2  0x0000155550c47121 in moldat (mode=0) at /home/paul/programs/mopac/src/moldat.F90:157
#3  0x0000155550dfff19 in mopac_api.mopac_api_initialize::mopac_initialize (system=...)
    at /home/paul/programs/mopac/src/interface/mopac_api_initialize.F90:319
#4  0x0000155550e046fb in mopac_api.mopac_api_operations::mopac_relax (system=...,
    state=..., properties=...)
    at /home/paul/programs/mopac/src/interface/mopac_api_operations.F90:86
#5  0x00000000004013b4 in run_optimization () at acetylene_opt.c:64
#6  0x000000000040140d in main () at acetylene_opt.c:77
```

here the code to reproduce:
```c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "mopac.h"

double run_optimization() {
    struct mopac_system system;
    struct mopac_state state;
    struct mopac_properties properties;
    
    memset(&system, 0, sizeof(struct mopac_system));
    memset(&state, 0, sizeof(struct mopac_state));
    memset(&properties, 0, sizeof(struct mopac_properties));
    
    system.natom = 4;
    system.natom_move = 4;
    system.charge = 0;
    system.spin = 0;
    system.model = 0;
    system.epsilon = 1.0;
    system.nlattice = 0;
    system.nlattice_move = 0;
    system.pressure = 0.0;
    system.tolerance = 1;
    system.max_time = 3600;
    
    system.atom = (int*)malloc(system.natom * sizeof(int));
    system.coord = (double*)malloc(3 * system.natom * sizeof(double));
    
    if (!system.atom || !system.coord) {
        return -1.0;
    }
    
    system.atom[0] = 6;  // C
    system.atom[1] = 6;  // C
    system.atom[2] = 1;  // H
    system.atom[3] = 1;  // H
    
    // C1 at origin
    system.coord[0] = 0.0;   // x
    system.coord[1] = 0.0;   // y
    system.coord[2] = 0.0;   // z
    
    // C2
    system.coord[3] = 1.20;  // x (C≡C triple bond)
    system.coord[4] = 0.0;   // y
    system.coord[5] = 0.0;   // z
    
    // H1 (bonded to C1)
    system.coord[6] = -1.06; // x (C-H bond)
    system.coord[7] = 0.0;   // y
    system.coord[8] = 0.0;   // z
    
    // H2 (bonded to C2)
    system.coord[9] = 2.26;  // x (1.20 + 1.06)
    system.coord[10] = 0.0;  // y
    system.coord[11] = 0.0;  // z

    state.mpack = 0;
    state.uhf = 0;
    state.pa = NULL;
    state.pb = NULL;
    
    mopac_relax(&system, &state, &properties);
    
    double heat = properties.heat;
    
    free(system.atom);
    free(system.coord);
    destroy_mopac_properties(&properties);
    destroy_mopac_state(&state);
    
    return heat;
}

int main() {
    double heat = run_optimization();
    printf("%.4f\n", heat);
    return 0;
}
```
happened with the following molecules: ['C=O', 'C#C', 'N(=O)O', 'N', 'C(#N)C#N', 'C(=N)=O', 'OO']

any obvious errors in my code that would cause this? printing na in gdb yields [0, 0, 0, 0], which is wanted behavior when defining the geometry through cartesian coordinates as i understand?

```c
if(size(na) == 4) call exit(1)
        do i = 4, natoms
          if (na(i) == 0) then
            coord(:,i) = geo(:,i)  ! Coordinate is already Cartesian
          end if
        end do
```

**Comment by godotalgorithm** [comment]
You aren't doing anything wrong. This is a bug that I need to fix. During normal operations, MOPAC pads out a lot of array sizes, so the `na` array will always be larger than 4 for any molecule and this barrier will never trigger. The API has partially re-implemented the initialization phase of MOPAC, and I did not pad out arrays because that shouldn't be necessary. I'll talk to the original MOPAC developer about this before I take any action - this barrier and the array padding may be a crude old fix for some other problems that I am not presently aware of.

You should be able to just take out that line for now and it should be fine. You are correct in that `gmetry` converts the input atomic coordinates into Cartesian coordinates, so it should do nothing if the input was already in Cartesian coordinates. `na` only has a nonzero value for atoms that are specified in internal coordinates (bond length, bond angle, and dihedral angle). The new API only supports Cartesian coordinates, so `na` should be all zeros.

I'll release a patch when I fix this and any related underlying issues.

**Comment by Acetylsalicylsaeure** [comment]
i see, thanks for the info! in the meantime, simply adding a dummy atom of type 99 to systems with natom==4 should not influence the calculation, right?

**Comment by godotalgorithm** [comment]
I don't have enough personal experience with dummy atoms to attest to this, but it probably shouldn't affect anything other than the size of some output arrays. MOPAC accounts for physical and unphysical atoms in a slightly different way, and I was mindful of that when implementing the API, but I did not do thorough testing with dummy atoms as they aren't really an intended use case.

I discussed the offending code with the original developer, and I will remove it today and release a patch. I'd like the new API to be as bug-free as possible, and resolving any API bugs are a high priority for me right now.
